GMG/GMG/Scenes/ChordProgress/ChordProgressView.swift

```diff
@@ -0,0 +1,515 @@
+                            chordCellAction: chordCellAction
+                        )
+                    }
+                }
+                .safeAreaPadding(Spacing.md)
+                .safeAreaPadding(.bottom, 128)
+            }
+            .mask {
+                VStack(spacing: .zero) {
+                    LinearGradient(
+                        stops: [
+                            Gradient.Stop(color: .clear, location: 0.0),
+                            Gradient.Stop(color: .white, location: 1.0),
+                        ],
+                        startPoint: .top,
+                        endPoint: .bottom
+                    )
+                    .frame(height: Spacing.md)
+                    Color.white
+                }
+                .ignoresSafeArea()
+            }
+        }
+    }
+
+    struct Segment: View {
+        let index: Int
+        let totalDuration: TimeInterval
+        let chordCells: [ChordCell]
+        let segmentDuration: TimeInterval
+        let currentChordCell: ChordCell?
+        let chordCellAction: (ChordCell) -> Void
+
+        private var segmentStartTime: TimeInterval {
+            TimeInterval(index) * segmentDuration
+        }
+
+        private var segmentEndTime: TimeInterval {
+            segmentStartTime + segmentDuration
+        }
+
+        private var filteredChordCells: [ChordCell] {
+            var filteredChordCells: [ChordCell] = chordCells.filter {
+                chordCell in
+                segmentStartTime <= chordCell.startTime
+                    && chordCell.startTime < segmentEndTime
+            }
+
+            /// 현재 세그먼트에 코드가 없을 경우 추가
+            if filteredChordCells.isEmpty,
+                let previousChordCell: ChordCell = chordCells.filter({
+                    chordCell in
+                    chordCell.startTime <= segmentStartTime
+                }).last
+            {
+                filteredChordCells = [previousChordCell]
+            }
+
+            /// 현재 세그먼트에 이전 코드 셀이 남아 있는 경우 추가
+            if let firstStartTime = filteredChordCells.first?.startTime,
+                firstStartTime - segmentStartTime > 0
+            {
+                if let previousChordCell: ChordCell = chordCells.filter({
+                    chordCell in
+                    chordCell.startTime <= segmentStartTime
+                }).last {
+                    filteredChordCells =
+                        [previousChordCell] + filteredChordCells
+                } else {
+                    filteredChordCells =
+                        [
+                            ChordCell(
+                                chord: nil,
+                                chordCandidates: [],
+                                startTime: segmentStartTime
+                            )
+                        ] + filteredChordCells
+                }
+            }
+
+            /// 마지막 세그먼트에서 오디오 길이보다 작을 경우
+            if index == Int(floor(totalDuration / 5))
```

Comment:
이 부분 5로 고정해놓으신 이유가 있으실까요? 따로 없다면 segmentDuration으로 변경해서 적용하면 좋을 것 같습니다
```suggestion
            if index == Int(floor(totalDuration / segmentDuration))
```